### PR TITLE
Support view comment in CREATE VIEW

### PR DESCRIPTION
**Pull request by liangxia2006**
Previous discussions are located at #8941

### CODE REVIEW

File: presto-docs/src/main/sphinx/sql/create-view.rst [hunk 1]
```diff
@@ -29,9 +31,11 @@ Create a simple view ``test`` over the ``orders`` table::
     SELECT orderkey, orderstatus, totalprice / 2 AS half
     FROM orders
 
-Create a view ``orders_by_date`` that summarizes ``orders``::
+Create a view ``orders_by_date`` that summarizes ``orders``, adding a view comment::
```
Comment (rongrong) [inline presto-docs/src/main/sphinx/sql/create-view.rst]: maybe change it to "with a comment"?

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java [hunk 2]
```diff
@@ -1094,7 +1094,8 @@ public void dropView(ConnectorSession session, SchemaTableName viewName)
                 views.put(schemaTableName, new ConnectorViewDefinition(
                         schemaTableName,
                         Optional.ofNullable(table.get().getOwner()),
-                        decodeViewData(table.get().getViewOriginalText().get())));
+                        decodeViewData(table.get().getViewOriginalText().get()),
```
Comment (rongrong) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java]: Why is this needed here?

File: presto-main/src/main/java/com/facebook/presto/execution/CreateViewTask.java [hunk 3]
```diff
@@ -90,7 +90,7 @@ public String explain(CreateView statement, List<Expression> parameters)
 
         String data = codec.toJson(new ViewDefinition(sql, session.getCatalog(), session.getSchema(), columns, Optional.of(session.getUser())));
 
-        metadata.createView(session, name.getCatalogName(), new ConnectorViewDefinition(name.asSchemaTableName(), Optional.empty(), data), statement.isReplace());
+        metadata.createView(session, name.getCatalogName(), new ConnectorViewDefinition(name.asSchemaTableName(), Optional.empty(), data, statement.getComment()), statement.isReplace());
```
Comment (rongrong) [inline presto-main/src/main/java/com/facebook/presto/execution/CreateViewTask.java]: I think this change should go with the next diff.

File: presto-main/src/main/java/com/facebook/presto/metadata/ViewInfo.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+package com.facebook.presto.metadata;
+
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public final class ViewInfo
+{
+    private final ViewDefinition definition;
+    private final Optional<String> comment;
```
Comment (rongrong) [inline presto-main/src/main/java/com/facebook/presto/metadata/ViewInfo.java]: Sorry I might be missing some context, but why not just add comment to ViewDefinition?

File: presto-spi/src/main/java/com/facebook/presto/spi/ConnectorViewDefinition.java [hunk 5]
```diff
@@ -22,12 +22,14 @@
     private final SchemaTableName name;
     private final Optional<String> owner;
     private final String viewData;
+    private final Optional<String> comment;
 
-    public ConnectorViewDefinition(SchemaTableName name, Optional<String> owner, String viewData)
+    public ConnectorViewDefinition(SchemaTableName name, Optional<String> owner, String viewData, Optional<String> comment)
```
Comment (rongrong) [inline presto-spi/src/main/java/com/facebook/presto/spi/ConnectorViewDefinition.java]: This change seems to better fit with the next diff.

File: presto-main/src/main/java/com/facebook/presto/metadata/ViewInfo.java (hunk 4 above)
Comment (electrum) [reply presto-main/src/main/java/com/facebook/presto/metadata/ViewInfo.java]: ViewDefinition is an opaque serialized object given to the connector. We want the comment to be a connector visible field.

File: presto-spi/src/main/java/com/facebook/presto/spi/ConnectorViewDefinition.java (hunk 5 above)
Comment (electrum) [reply presto-spi/src/main/java/com/facebook/presto/spi/ConnectorViewDefinition.java]: This is the correct diff.

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java (hunk 2 above)
Comment (liangxia2006) [reply presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java]: This is the original behavior. When creating a view in HiveMetadata, the view data is encoded `setViewOriginalText(Optional.of(encodeViewData(viewData)))` ; so that here it needs to be decoded.

Comment (rongrong) [reply presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java]: Oops I miss matched the parenthesis and understood it wrong...

File: presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorMetadata.java [hunk 6]
```diff
@@ -810,29 +810,29 @@ public boolean supportsMetadataDelete(ConnectorSession session, ConnectorTableHa
     }
 
     @Override
-    public void createView(ConnectorSession session, SchemaTableName viewName, String viewData, boolean replace)
+    public void createView(ConnectorSession session, ConnectorViewDefinition definition, boolean replace)
     {
-        String schemaName = viewName.getSchemaName();
-        String tableName = viewName.getTableName();
+        String schemaName = definition.getName().getSchemaName();
```
Comment (liangxia2006) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorMetadata.java]: This is wrong. I will update it soon.

**Comment by liangxia2006** [comment]
Sorry that the previous comments may have been lost due to my force push. Here are the previous comment from David.
1) Coding styles: 
  1a) Create local variables viewName and viewData so that rest of the code don't need to change due to refactoring to ConnectorViewDefinition;
  1b) Wrap all or none.
2) Keep the old sql formatting without comment.
3) Add a smoke test for hive show create view.

**Comment by liangxia2006** [comment]
Hi @electrum , please let me know if you have more comments. 
For the Travis CI run, last time it failed on job 25842.7, but this time it failed on a different job 25842.9. They are all unrelated to my commits. Not sure whether this is something that I need to rerun or fix.

**Comment by shixuan-fan** [comment]
It might be a flaky test. Let me rerun it to see if it will work this time. @liangxia2006
